### ISSUE TITLE
Can use redis-benchmark to test redis3.2.8?

### ISSUE THREAD

**Issue by fantastic2085** [F-Stack/f-stack#358]

**Comment by fantastic2085** [comment]
redis-benchmark:main->aeCreateEventLoop->aeApiCreate(ae_ff_kqueue.c) below:
 state->kqfd = ff_kqueue();  ----> Segment error

With the relevant "EAL:no free hugepages reported in hugepages-1048576KB"?

**Comment by cxl-ironman** [comment]
It seems the redis-benxhmark inside f-stack did no port to f-stack yet. You can use native version to test in another server.

**Comment by fantastic2085** [comment]
It seems the redis-benxhmark inside f-stack did no port to f-stack yet. You can use native version to test in another server.

Could you tell me more about it? @cxl-ironman  
My goal is to use DPDK on both the receiving end and the sending end

**Comment by fantastic2085** [comment]
> It seems the redis-benxhmark inside f-stack did no port to f-stack yet. You can use native version to test in another server.

My goal is to use DPDK on both the receiving end and the sending end

**Comment by cxl-ironman** [comment]
The redis-benchmark code in f-stack is the same as the redis official version. However, when it use kqueue api, it include ae_ff_kqueue.c(see line 50 in src/ae.c),  which needs ff_init(), ff_run() and any other f-stack initialization. I think this is an incompatible problem.

You can download the official version of redis, complied and then use redis-benchmark to test on another server.

**Comment by fantastic2085** [comment]
> The redis-benchmark code in f-stack is the same as the redis official version. However, when it use kqueue api, it include ae_ff_kqueue.c(see line 50 in src/ae.c), which needs ff_init(), ff_run() and any other f-stack initialization. I think this is an incompatible problem.
> You can download the official version of redis, complied and then use redis-benchmark to test on another server.

Thank you@cxl-ironman 
I have downloaded version 3.2.8 from redis github and used redis-benchmark to test!  It works.
Howerver，My goal is to use DPDK on both the receiving end and the sending end！

**Comment by cxl-ironman** [comment]
> Thank you@cxl-ironman
> I have downloaded version 3.2.8 from redis github and used redis-benchmark to test! It works.
> Howerver，My goal is to use DPDK on both the receiving end and the sending end！

Do you want to use f-stack to send and receive packets or just want to implement redis client with f-stack?

**Comment by fantastic2085** [comment]
Redis-benchmark based on fstack is not working, can I understand that?@cxl-ironman

**Comment by fantastic2085** [comment]
> > Thank you@cxl-ironman
> > I have downloaded version 3.2.8 from redis github and used redis-benchmark to test! It works.
> > Howerver，My goal is to use DPDK on both the receiving end and the sending end！
> 
> Do you want to use f-stack to send and receive packets or just want to implement redis client with f-stack?

@cxl-ironman  Using DPDK to test redis, but, Redis-benchmark based on fstack is not working！

**Comment by cxl-ironman** [comment]
> Redis-benchmark based on fstack is not working, can I understand that?@cxl-ironman

yes, may be you can write a client yourself using f-stack and hiredis to test.

**Comment by fantastic2085** [comment]
redis3.2.8 server use fstack which based on DPDK， I thought redis-benchmark also use fstack，that means  to use DPDK on both the receiving end and the sending end！but, Redis-benchmark based on fstack is not working！！

**Comment by cxl-ironman** [comment]
> redis3.2.8 server use fstack which based on DPDK， I thought redis-benchmark also use fstack，that means to use DPDK on both the receiving end and the sending end！but, Redis-benchmark based on fstack is not working！！

yes,  I think redis-benchmark and redis-cli in f-stack is not using DPDK.

**Comment by ericgong1993** [comment]
> It seems the redis-benxhmark inside f-stack did no port to f-stack yet. You can use native version to test in another server.

I donnot understand，I have the same problem ，I cannot test with redis-benchmark

**Comment by ericgong1993** [comment]
> > The redis-benchmark code in f-stack is the same as the redis official version. However, when it use kqueue api, it include ae_ff_kqueue.c(see line 50 in src/ae.c), which needs ff_init(), ff_run() and any other f-stack initialization. I think this is an incompatible problem.
> > You can download the official version of redis, complied and then use redis-benchmark to test on another server.
> 
> Thank you@cxl-ironman
> I have downloaded version 3.2.8 from redis github and used redis-benchmark to test! It works.
> Howerver，My goal is to use DPDK on both the receiving end and the sending end！

你好 我也遇到这个问题了，但是我在客户端下载官方的redis3.2.8版本来进行redis-benchmark压测，无法正常压测呢，请问你最后是如何解决的？

**Comment by jfb8856606** [comment]
To summarize the discussion above:

**F-Stack only ports the Redis server** (`redis-server`) to use the F-Stack network stack. The `redis-benchmark` and `redis-cli` tools are **not ported to F-Stack** — they remain the same as the official Redis versions. Calling `ff_kqueue()` from redis-benchmark will segfault because `ff_init()` / `ff_run()` and other DPDK initialization are missing.

**How to test F-Stack Redis:**
- Use the official (unmodified) `redis-benchmark` from a separate machine, connecting through the normal network to the F-Stack Redis server
- If you need both ends on DPDK, you would need to write a custom client using F-Stack + hiredis

This applies to both the old Redis 3.2.8 and the current Redis 6.2.6 in F-Stack.

Closing as this is a known limitation and has been thoroughly discussed.
